File: backend/models/TeacherProfile.js
// models/TeacherProfile.js
const mongoose = require("mongoose");

const TeacherProfileSchema = new mongoose.Schema(
  {
    userId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "User",
      required: true,
      unique: true
    },

    university: { type: String, default: "" },
    department: { type: String, default: "" },
    subjects: { type: [String], default: [] },
    classLevels: { type: [String], default: [] },

    jobTitle: { type: String, default: "" }, // Lecturer / Tutor / etc.

    // Salary expectation range
    expectedSalaryMin: { type: Number, default: 0 },
    expectedSalaryMax: { type: Number, default: 0 },

    location: {
      type: {
        type: String,
        enum: ["Point"],
        default: "Point"
      },
      coordinates: {
        type: [Number], // [lng, lat]
        default: [0, 0]
      },
      city: { type: String, default: "" },
      area: { type: String, default: "" }
    },

    availability: {
      days: { type: [String], default: [] },
      timeRange: { type: String, default: "" }
    },

    about: { type: String, default: "" },

    // NID (National ID) verification
    nidCardImageUrl: {
      type: String,
      default: null
    },
    isNidVerified: {
      type: Boolean,
      default: false
    },

    isVerified: {
      type: Boolean,
      default: false
    },

    // ⭐ Rating fields
    ratingAverage: {
      type: Number,
      default: 0
    },
    ratingCount: {
      type: Number,
      default: 0
    }
  },
  { timestamps: true }
);

// 2dsphere index for geo queries
TeacherProfileSchema.index({ location: "2dsphere" });

module.exports = mongoose.model("TeacherProfile", TeacherProfileSchema);

File: lib/src/core/models/teacher_profile.dart
import 'location_model.dart';

class TeacherProfile {
  final String id;
  final List<String> subjects;
  final List<String> classLevels;
  final int expectedSalaryMin;
  final int expectedSalaryMax;
  final String university;
  final String department;
  final String jobTitle;
  final String about;
  final bool isVerified;
  final double ratingAverage;
  final int ratingCount;
  final LocationPoint location;

  TeacherProfile({
    required this.id,
    required this.subjects,
    required this.classLevels,
    required this.expectedSalaryMin,
    required this.expectedSalaryMax,
    required this.university,
    required this.department,
    required this.jobTitle,
    required this.about,
    required this.isVerified,
    required this.ratingAverage,
    required this.ratingCount,
    required this.location,
  });

  factory TeacherProfile.fromJson(Map<String, dynamic> json) {
    return TeacherProfile(
      id: json["_id"] ?? "",
      subjects: List<String>.from(json["subjects"] ?? []),
      classLevels: List<String>.from(json["classLevels"] ?? []),
      expectedSalaryMin: (json["expectedSalaryMin"] ?? 0).toInt(),
      expectedSalaryMax: (json["expectedSalaryMax"] ?? 0).toInt(),
      university: json["university"] ?? "",
      department: json["department"] ?? "",
      jobTitle: json["jobTitle"] ?? "",
      about: json["about"] ?? "",
      isVerified: json["isVerified"] ?? false,
      ratingAverage: (json["ratingAverage"] ?? 0).toDouble(),
      ratingCount: (json["ratingCount"] ?? 0).toInt(),
      location: LocationPoint.fromJson(json["location"]),
    );
  }
}
